contrib/win32/win32compat/w32fd.c

```diff
@@ -1069,60 +1070,154 @@ spawn_child_internal(char* cmd, char *const argv[], HANDLE in, HANDLE out, HANDL
 	else
 		cmdline_len += (DWORD)strlen(path) + 1 + 2;
 
-	if (argv) {
+		if (argv) {
 		t1 = argv;
-		while (*t1)
-			cmdline_len += (DWORD)strlen(*t1++) + 1 + 2;
+		while (*t1) {
+			if (!escape)
```

Comment:
Recommend pulling all the commandline building logic into a separate utility helper and writing an exhaustive set of unit tests.

Reply:
the is for the case when the module path is added in this function. We need to add quotes

Reply:
separate the utility helper out. 
Will add unit tests.